syntax = "proto3";
package refine;

message Clip {
  // Numpy .npy bytes of shape [T, 3, 32, 32] float32 (values 0..1)
  bytes frames_npy = 1;

  // For logging/traceback only (optional)
  int32 start_index = 2;
  int32 end_index = 3;
}

message RefineResult {
  string label = 1;       // "Suspicious" or "Normal"
  float confidence = 2;   // 0..1
  string detail = 3;      // free-form info (timings, model name)
}

service RefineService {
  rpc Refine(Clip) returns (RefineResult);
}